### DISCUSSION TITLE
What functionality is Vitepress' 18n providing?

### DISCUSSION THREAD

**Discussion by CodyBontecou** [vuejs/vitepress#1911] [Q&A] [answered]
I've read the [docs](https://vitepress.vuejs.org/guide/i18n#internationalization) and am still trying to figure out what features Vitepress' i18n is providing. 

As far as I can tell, I'm simply providing the files in separate directories specific to the language. Does Vitepress redirect users to the appropriate page based on their browser's default language?

**Comment by brc-dd** [comment, marked as answer]
> Does Vitepress redirect users to the appropriate page based on their browser's default language?

That needs to be configured in your server. (VitePress is just a SSG, having no server after build.) For example, on Netlify, it is configurable like mentioned in this section: https://vitepress.vuejs.org/guide/i18n#separate-directory-for-each-locale

**Reply by CodyBontecou** [reply]
Yup, I saw that section. I guess I'm just curious why I should be using the `locale` configuration.

**Reply by brc-dd** [reply]
For customizing the parts of theme based on which locale directory (/zh/, /en/) you're on.

**Reply by CodyBontecou** [reply]
Ah gotcha. For example, the sidebar and navbar?

**Reply by brc-dd** [reply]
Yeah that, and rest of the placeholders (like search, that "on this page" label, "previous page"/"next page" thing, ...). Nearly the whole theme. Also, you can have html attributes/head config, etc. based on that. It's basically for having different stuff for different locales if you want.

**Reply by CodyBontecou** [reply]
Okay, good to know! Thank you for clarifying.
